### DISCUSSION TITLE
How to separate local executors?

### DISCUSSION THREAD

**Discussion by BeyondEvil** [nrwl/nx#9831] [General]
When running `run` command (perhaps others as well, haven't tested), I get the same warning twice:

```sh
Skipping project found at tools/executors/migrate since project tools-executors already exists at tools/executors/gen-models! Specify a unique name for the project to allow Nx to differentiate between the two projects.
```

Directory structure:

```sh
tools/executors
├── gen-models
│   ├── executor.json
│   ├── impl.ts
│   ├── package.json
│   └── schema.json
└── migrate
    ├── executor.json
    ├── impl.ts
    ├── package.json
    └── schema.json
```

They each have a unique `id` in `schema.json`.

What else is needed?

**Comment by sandersrd33** [comment]
Did you find any answers or workarounds for this? Seeing the same problem. But I'm using pnpm on Windows so not sure if it might be related to that.

**Reply by BeyondEvil** [reply]
> Did you find any answers or workarounds for this? Seeing the same problem. But I'm using pnpm on Windows so not sure if it might be related to that.

Unfortunately not 😔

**Comment by AgentEnder** [comment]
Hey @BeyondEvil! This error message comes from project-inference finding 2 projects that have the same name when `workspace.json` doesn't exist. The package.json files in your executors contain the same name most likely.

**Reply by BeyondEvil** [reply]
Thanks @AgentEnder 

However, we _do_ have a `workspace.json` file. Neither have a name set in `package.json` tho.

**Reply by AgentEnder** [reply]
If you have a workspace.json, something weird is going on. Can you create a minimal reproduction and link it?

**Reply by BeyondEvil** [reply]
TS is really not my strong suit, I'm the poor DevOps guy. So I'm not sure I can. :(

**Comment by MFry** [comment]
I was getting a similar issue in versions `15.2.1` and `15.7.2` when building the docker containers.
Not sure why this started randomly happening, but `nx run [project]:build:production` would cause an endless loop of the message above

```
Skipping project found at apps/[project] since project [project] already exists at .! Specify a unique name for the project to allow Nx to differentiate between the two projects.
> nx run [project]:build:production
```

running `nx run [project]:build` instead prevented the endless loop from occurring. 

Note that `15.2.1` had all unique named projects in `workspace.json` and `15.7.2` migration removed the `workspace.json` entirely with still uniquely named projects.

**Reply by AgentEnder** [reply]
Your root package.json file has a name that matches one of your projects, rename it.
